shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -461,19 +492,56 @@ - (void)dispatchMouseEvent:(NSEvent*)event phase:(FlutterPointerPhase)phase {
   }
 }
 
+- (void)propagateKeyEvent:(NSEvent*)event ofType:(NSString*)type {
+  if ([type isEqual:@"keydown"]) {
+    for (NSResponder* responder in self.additionalKeyResponders) {
+      if ([responder respondsToSelector:@selector(keyDown:)]) {
+        [responder keyDown:event];
```

Comment:
Hm, we're going to need to change the interface on add/removeKeyResponder from NSResponder to something custom, because we need to get a boolean back from this responder, and gate the call below on that. With this change as written, all key events handled by the text input system would incorrectly propagate to the next responder as well.

Reply:
OK, I introduced a new class for this that only handles key up and key down.